sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -55,13 +60,18 @@ case class CreateViewCommand(
   require(tableDesc.tableType == CatalogTableType.VIEW)
   require(tableDesc.viewText.isDefined)
 
-  private val tableIdentifier = tableDesc.identifier
-
   if (allowExisting && replace) {
     throw new AnalysisException(
       "It is not allowed to define a view with both IF NOT EXISTS and OR REPLACE.")
   }
 
+  // Temporary view names should NOT contain database prefix like "database.table"
+  if (isTemporary && tableDesc.identifier.database.isDefined) {
+    val database = tableDesc.identifier.database.get
+    throw new AnalysisException(
```

Comment:
This is to be consistent with DataSet API. When registering a temp table, it will remove the database prefix, here is the code:
https://github.com/apache/spark/blob/6ba17cd147277a20a7fbb244c040e694de486c36/sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala#L533